pkgs/games/quakespasm/default.nix

```diff
@@ -14,10 +26,22 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ copyDesktopItems ];
   buildInputs = [
-    gzip SDL libvorbis libmad
-  ];
+    gzip libvorbis libmad (if useSDL2 then SDL2 else SDL)
+  ] ++ lib.optionals stdenv.isDarwin macFrameworks;
 
-  buildFlags = [ "DO_USERDIRS=1" ];
+  CFLAGS = lib.optionalString stdenv.isDarwin "-DGL_SILENCE_DEPRECATION";
+  buildFlags = [ "DO_USERDIRS=1" ]
+    ++ lib.optional useSDL2 "USE_SDL2=1";
+
+  # NOTE: for macOS, we're using generic Makefile instead of Makefile.darwin, since we are not going to build an app bundle.
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    substituteInPlace Makefile --replace "-lGL" "${lib.concatMapStringsSep " " (e: "-Wl,-framework,${e}") macFrameworkNames}"
+  '' + lib.optionalString (useSDL2 && stdenv.isDarwin) ''
+    # SDL 2.x does not replace main() and since we're not going to build an app bundle for macOS, then there is no launcher to call SDL_main.
+    substituteInPlace main_sdl.c --replace "#define main SDL_main" ""
```

Comment:
and this can be moved to only check if we are on darwin.

Reply:
This will be dropped